presto-hive/src/test/java/com/facebook/presto/hive/TestHiveSkipEmptyFiles.java

```diff
@@ -0,0 +1,392 @@
+        executeCreationTestAndDropCycle(queryFailRunner, tableName, getResourceUrl(tableName), true,
+                ".* is not a valid Parquet File");
+        deleteMetadata(resourcesLocation, false);
+    }
+
+    @Test
+    public void testSkipEmptyFilesBucketSuccessful()
+            throws Exception
+    {
+        String tableName = "skip_empty_files_bucket_success";
+        File resourcesLocation = generateBucketedMetadata(queryBucketRunner, tableName, false);
+        checkBucketedResult(queryBucketRunner, tableName, false, null);
+        deleteMetadata(resourcesLocation, false);
+    }
+
+    @Test
+    public void testSkipEmptyFilesBucketInsertFileFail()
+            throws Exception
+    {
+        String tableName = "skip_empty_files_bucket_insert_fail";
+        File resourcesLocation = generateBucketedMetadata(queryBucketFailRunner, tableName, false);
+        checkBucketedResult(queryBucketFailRunner, tableName, true, ".* is corrupt.* does not match the standard naming pattern, and the number of files in the directory .* does not match the declared bucket count.*");
+        deleteMetadata(resourcesLocation, false);
+    }
+
+    @Test
+    public void testSkipEmptyFilesBucketReplaceFileFail()
+            throws Exception
+    {
+        String tableName = "skip_empty_files_bucket_replace_fail";
+        File resourcesLocation = generateBucketedMetadata(queryBucketFailRunner, tableName, true);
+        checkBucketedResult(queryBucketFailRunner, tableName, true, ".* is not a valid Parquet File");
+        deleteMetadata(resourcesLocation, false);
+    }
+
+    /**
+     * Obtains the external location from the local resources directory of the project
+     *
+     * @param tableName a {@link String} containing the directory name to search for
+     * @return a {@link String} with the external location for the given table_name
+     */
+    private static String getResourceUrl(String tableName)
+    {
+        URL resourceUrl = TestHiveSkipEmptyFiles.class.getClassLoader().getResource(tableName);
+        if (resourceUrl == null) {
+            throw new RuntimeException("Cannot find resource path for table name: " + tableName);
+        }
+        return resourceUrl.toString();
+    }
+
+    /**
+     * Tries a table with the configuration property desired. If succeeds, tests the output.
+     * Finally, it drops the table.
+     *
+     * @param queryRunner a {@link QueryRunner} with the desired configuration properties
+     * @param tableName a {@link String} containing the desired table name
+     * @param shouldFail {@code true} if the table creation should fail, {@code false} otherwise
+     * @param errorMessage a {@link String} containing the expected error message. Will be checked if {@code shouldFail} is {@code true}
+     */
+    private void checkBucketedResult(DistributedQueryRunner queryRunner, String tableName, boolean shouldFail, @Language("RegExp") String errorMessage)
```

Comment:
boolean parameters to change behavior are an antipattern that increases cyclomatic complexity. Use two methods, one for things that should fail and one for things that should pass.

Reply:
Fixed